core/tx_list.go

```diff
@@ -279,15 +279,23 @@ func (l *txList) Add(tx *types.Transaction, priceBump uint64) (bool, *types.Tran
 	// If there's an older better transaction, abort
 	old := l.txs.Get(tx.Nonce())
 	if old != nil {
-		// threshold = oldGP * (100 + priceBump) / 100
+		if old.FeeCapCmp(tx) >= 0 || old.TipCmp(tx) >= 0 {
```

Comment:
In this strategy, we can only replace the transaction with 10% feeCap bump and 10% tip bump. It might hurt the UX to some extend.

For example, if the user already specify a high enough feeCap but a barely enough tip. And he wants to speed up the inclusion, he has to bump the feeCap as well. If the basefee raises up, then the maximum possible payment is increased. 

Of course, bumping both factors is easiest. Not sure if it's worthwhile to introduce addition complexity

Reply:
This has been discussed at the 1559 dev channel, right now this replacement condition seems to be the best choice.